renku/domain_model/dataset.py

```diff
@@ -318,6 +318,7 @@ class Dataset(Persistent):
     """Represent a dataset."""
 
     date_modified: Optional[datetime] = None  # type: ignore
+    storage: Optional[str] = None
```

Comment:
Do we really need to have `storage` as both a class and instance variable? Isn't it simpler to just have the instance variable?

But maybe I am missing something. This also seems to happen with a few other class/instance variables on dataset.

Reply:
We need the class variable for backward compatibility: When we load an object from an older project that doesn't have the `storage` property, the class variable makes sure that this property is set to `None`. The library that we use to store Renku metadata (i.e. ZODB/zope/persistent) doesn't call the `__init__` method when deserializing an object (it creates an empty object and calls `__setstate__` upon the first access to the object). So, we need the class variable for every new attribute that we add to our models.